src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -29,50 +28,76 @@ object WdlExpression {
 
   val preEvaluableFunctions_SingleParameter: Seq[String] = Seq("read_int", "read_string")
 
+  // Function calls which can be called from within other function calls.
+  // e.g. read_int(stdout()) is fine.
+  val allowedInnerFunctionCalls = Seq[String]("stdout", "stderr")
+
   def functionCall(a: Ast): Boolean = a.getName == "FunctionCall"
 
   def binaryOperator(a: Ast): Boolean = binaryOperators.contains(a.getName)
   
-  def unaryOperator(a: Ast): Boolean = unaryOperators.contains((a.getName))
-  
+  def unaryOperator(a: Ast): Boolean = unaryOperators.contains(a.getName)
+
+  def getFunctionName(a: Ast): String = a.getAttribute("name").asInstanceOf[Terminal].getSourceString
+
   def functionCallWithOneFileParameter(a: Ast): Boolean = (
     functionCall(a)
     && a.getAttribute("params").asInstanceOf[AstList].asScala.toVector.size == 1
-    && preEvaluableFunctions_SingleParameter.contains(a.getAttribute("name").asInstanceOf[Terminal].getSourceString))
-  
+    && preEvaluableFunctions_SingleParameter.contains(getFunctionName(a)))
+
+  /**
+   * Checks whether any nodes in the AST are function calls.
+   */
+  def containsFunctionCalls(ast: AstNode): Boolean = {
```

Comment:
I think this would end up looking better when used if it was a method on `AstNode` (via an implicit value class if it doesn't make sense to directly put it in there). Then you'd have things like `innerAst.containsFunctioncalls` instead of `containsFunctionCalls(innerAst)`